spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::UsersController, type: :controller do
+  describe 'POST#create' do
+    context 'given valid parameters' do
+      it 'creates a new user' do
+        expect do
+          post :create, params: { username: 'nimblehq', password: 'password', password_confirmation: 'password' }
+        end.to change(User, :count).by(1)
+      end
+
+      it 'returns created status' do
+        post :create, params: { username: 'nimblehq', password: 'password', password_confirmation: 'password' }
+
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'returns JSON API response body' do
+        post :create, params: { username: 'nimblehq', password: 'password', password_confirmation: 'password' }
+
+        response_body = JSON.parse(response.body, symbolize_names: true)
+
+        expect(response_body[:data][:type]).to eql('user')
+        expect(response_body[:data][:attributes][:username]).to eql('nimblehq')
+      end
+    end
+
+    context 'given invalid parameters' do
+      it 'does NOT create a new user when username is nil' do
```

Comment:
we don't need to check every single case of an invalid parameter. It should be done in the User model spec, e.g.

```ruby
...
describe 'Validations' do
  subject { Fabricate(:user) }

  it { is_expected.to validate_presence_of(:username) }
  it { is_expected.to validate_presence_of(:password) }
...
end
```

and in this controller test, we can just test a generic case of `invalid paramters` only where we have parameters with one/some of them is invalid is enough.

Reply:
Fixed in https://github.com/gutakk/ruby-google-scraper/pull/17